sklearn/preprocessing/discretization.py

```diff
@@ -166,7 +166,7 @@ def _validate_n_bins(self, n_features, ignored):
         """
         orig_bins = self.n_bins
         if isinstance(orig_bins, numbers.Number):
-            if not isinstance(orig_bins, np.int):
+            if not isinstance(orig_bins, (np.int, np.integer)):
```

Comment:
At a glance, this seems not consistent with #10017. Seems that we are going to use `(numbers.Integral, np.integer)`?